frameworks/CSharp/aspnetcore/Benchmarks/Data/RavenDb.cs

```diff
@@ -0,0 +1,135 @@
+            var wstore = new DocumentStore
+            {
+                Urls = new[] { _connectionString },
+                Database = "world",
+            };
+
+            wstore.Conventions.FindCollectionName = type =>
+            {
+                if (typeof(WorldRaven).IsAssignableFrom(type))
+                    return "World";
+
+                return DocumentConventions.DefaultGetCollectionName(type);
+            };
+
+            var fstore = new DocumentStore
+            {
+                Urls = new[] { _connectionString },
+                Database = "fortunes",
+            };
+
+            fstore.Conventions.FindCollectionName = type =>
+            {
+                if (typeof(FortuneRaven).IsAssignableFrom(type))
+                    return "Fortune";
+
+                return DocumentConventions.DefaultGetCollectionName(type);
+            };
+
+            wstore.Initialize();
+            fstore.Initialize();
+
+            _worldStore = wstore;
+            _fortuneStore = fstore;
+        }
+
+        public async Task<WorldRaven> LoadSingleQueryRow()
+        {
+            using (var session = _worldStore.OpenAsyncSession())
+            {
+                return await session.LoadAsync<WorldRaven>(_identifiers[_random.Next(1, 10001)]);
+            }
+        }
+
+        public async Task<IEnumerable<WorldRaven>> LoadMultipleQueriesRows(int count)
+        {
+            using (var session = _worldStore.OpenAsyncSession())
+            {                
+                var ids = new string[count];
+
+                for (var i = 0; i < count; i++)
+                    ids[i] = _identifiers[_random.Next(1, 10001)];
+
+                var result = await session.LoadAsync<WorldRaven>(ids)                    ;
+                return result.Values;
+            }
+        }
+
+        public async Task<IEnumerable<WorldRaven>> LoadMultipleUpdatesRows(int count)
+        {
+            using (var session = _worldStore.OpenAsyncSession())
+            {
+                var ids = new string[count];
+
+                for (var i = 0; i < count; i++)
+                    ids[i] = _identifiers[_random.Next(1, 10001)];
+
+                var result = await session.LoadAsync<WorldRaven>(ids);
+                foreach (var pair in result)
+                {
+                    pair.Value.RandomNumber = _random.Next(1, 10001);
+                }
+
+                await session.SaveChangesAsync();
+
+                return result.Values;
+            }
+        }
+
+        public async Task<List<FortuneRaven>> LoadFortunesRows()
+        {
+            using (var session = _fortuneStore.OpenAsyncSession())
+            {
+                var list = new List<FortuneRaven>();
+
+                // TODO: We need to have an static index and stream it completely. 
+                var query = session.Advanced
+                                   .AsyncRawQuery<FortuneRaven>("from Fortune order by Id");
```

Comment:
I think you can drop the `order by Id` here, this is the natural sort order when querying by collection only.

Reply:
I was under the impression that it would be lexicographic instead. Gonna check that.

Reply:
It is lexical anyway

Reply:
Yes, you are right there... I may be able to simplify it further